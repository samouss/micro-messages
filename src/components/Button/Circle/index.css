.Circle {
  background-color: #0169d9;
  border: 2px solid #FFFFFF;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  box-shadow: 0px 0px 10px 0px rgba(0,0,0,0.5);
  color: #FFFFFF;
  font-size: 2rem;
}
