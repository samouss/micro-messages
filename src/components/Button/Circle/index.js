// @flow

import React from 'react';
import './index.css';

export type Props = {
  children: React.Element<*>,
};

const Circle = ({ children, ...props }: Props) => (
  <button
    {...props}
    styleName="Circle"
  >
    {children}
  </button>
);

export default Circle;
