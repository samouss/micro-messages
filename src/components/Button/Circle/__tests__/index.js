// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Circle from '../index';

describe('<Circle />', () => {
  const defaultProps = {
    onClick: () => {},
  };

  it('expect to render the component correctly', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Circle
        {...props}
      >
        <span>some content</span>
      </Circle>,
    );

    expect(component).toMatchSnapshot();
  });
});
