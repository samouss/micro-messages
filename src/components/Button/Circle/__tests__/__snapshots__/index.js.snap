// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> expect to render the component correctly 1`] = `
<button
  onClick={[Function]}
  styleName="Circle"
>
  <span>
    some content
  </span>
</button>
`;
